loadbalancer/fadein_test.go

```diff
@@ -190,6 +196,95 @@ func TestFadeIn(t *testing.T) {
 	testFadeIn(t, "consistent-hash, 9", newConsistentHash, fadeInDuration/2, fadeInDuration/3, fadeInDuration/4)
 }
 
+func closeToZero(arg int, max int) bool {
+	return arg < max/100
+}
+
+func closeToMax(arg int, max int) bool {
+	return arg > 90*max/100 && arg <= max
+}
+
+func testFadeInLoadBetweenOldEps(
+	t *testing.T,
+	name string,
+	algorithm func([]string) routing.LBAlgorithm,
+	endpointAges ...time.Duration,
+) {
+	t.Run(name, func(t *testing.T) {
+		old := 2 * fadeInDurationHuge
+		new := time.Duration(0)
+		ctx, eps := initializeEndpoints(endpointAges, fadeInDurationHuge)
+
+		a := algorithm(eps)
+		rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
+		nReqs := map[string]int{}
+		stop := time.After(3 * time.Second)
+
+		t.Log("test start", time.Now())
+		// Emulate the load balancer loop, sending requests to it with random hash keys
+		// over and over again till fadeIn period is over.
+	outer:
+		for {
+			ctx.Params[ConsistentHashKey] = strconv.Itoa(rnd.Intn(100000))
+			ep := a.Apply(ctx)
+			nReqs[ep.Host]++
+			select {
+			case <-stop:
+				break outer
+			default:
+			}
+		}
+
+		max := 0
+		for _, v := range nReqs {
+			if v > max {
+				max = v
+			}
+		}
+
+		for idx, ep := range eps {
+			if endpointAges[idx] == old && !closeToMax(nReqs[ep], max) {
+				t.Errorf("endpoint %+v is not close to max: %+v", ep, nReqs)
+			}
+			if endpointAges[idx] == new && !closeToZero(nReqs[ep], max) {
+				t.Errorf("endpoint %+v is not close to zero: %+v", ep, nReqs)
+			}
+		}
+	})
+}
+
+func TestFadeInLoadBetweenOldEps(t *testing.T) {
+	old := 2 * fadeInDurationHuge
+	testFadeInLoadBetweenOldEps(t, "consistent-hash", newConsistentHash, 0, old, old, old)
+	testFadeInLoadBetweenOldEps(t, "random", newRandom, 0, old, old, old)
+	testFadeInLoadBetweenOldEps(t, "round-robin", newRoundRobin, 0, old, old, old)
+}
+
+func testApplyEndsWhenAllEndpointsAreFading(
+	t *testing.T,
+	name string,
+	algorithm func([]string) routing.LBAlgorithm,
+	nEndpoints int,
+) {
+	// Initialize every endpoint with zero: every endpoint is new
+	endpointAges := make([]time.Duration, nEndpoints)
+
+	ctx, eps := initializeEndpoints(endpointAges, fadeInDurationHuge)
+
+	a := algorithm(eps)
+	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
+	ctx.Params[ConsistentHashKey] = strconv.Itoa(rnd.Intn(100000))
```

Comment:
As this parameter is used one time, you can use constant here (or remove at all).